lib/src/open_food_api_client.dart

```diff
@@ -109,6 +109,10 @@ class OpenFoodAPIClient {
         if (pos != -1) {
           key = key.substring(0, pos);
         }
+        final servingPost = key.indexOf('_serving');
+        if (servingPost != -1) {
+          key = key.substring(0, servingPost);
+        }
```

Comment:
It would be way much cleaner to loop on `PerSize.values` rather than testing on `_100g` and `_serving` that come out of nowhere.

Reply:
Agreed!